worker/src/services/IngestionService/index.ts

```diff
@@ -332,13 +373,26 @@ export class IngestionService {
     return {
       ...parsedObservationRecord,
       ...generationUsage,
-      event_ts: Math.max(
-        ...observationRecords.map((r) => r.event_ts),
-        clickhouseObservationRecord?.event_ts ?? -Infinity,
+      event_ts: this.getEventTimestamp(
+        clickhouseObservationRecord?.event_ts,
+        recordsToMerge.map((r) => r.event_ts ?? -Infinity),
       ),
     };
   }
 
+  /**
+   * Generates the event_ts from a list of events. If a clickhouse timestamp is provided, it will always be preferred
+   * as it's not mutable on the clickhouse side.
+   * Otherwise, we use the maximum timestamp. Either it should be provided, or always be the same, and using the
+   * max has the nice property, that we overwrite the default date of 1970-01-01.
+   */
+  private getEventTimestamp(
+    clickhouseTimestamp: number | undefined,
+    otherTimestamps: number[],
+  ): number {
+    return clickhouseTimestamp ?? Math.max(...otherTimestamps);
+  }
```

Comment:
@maxdeichmann Any concerns about this approach? This is a slight deviation from what we've discussed, as we're using the max now, but IMO that's safer and shouldn't make a difference for all well-behaved clients.